docs/src/main/mdoc/migration.md

```diff
@@ -0,0 +1,52 @@
+---
+layout: home
+title:  "migration guide"
+section: "Home"
+position: 7
+---
+
+## v0.18.0 Migraion Guide
```

Comment:
Do you think it's helpful to have this migration guide on a separate page? What if we put it on the Changes page?
http://typelevel.org/spire/extra_md/changes.html

Reply:
At first, I thought it would be helpful because there are some breaking change between 0.17.0 and 0.18.0 and users can easily find them (In general, users take it for granted that there are breaking API changes between major versions like 0.x.y and 1.0.0, but may not between minors.), but after your comment and thinking twice, the migration guide in separate page is not good idea because we need to maintain CHANGES.md, migration.md and release notes at github that are almost the same. Thank you for pointing this out.

Reply:
> maintain CHANGES.md, migration.md and release notes at github that are almost the same.

Hmm, you're right, that seems like a lot of duplication. Actually, why _aren't_ they the same? Is there something missing from the release notes, maybe we should add that there instead? Since many users may get their information from there without checking the website I think.

I think `CHANGES.md` was important before GitHub had "releases" feature. But now we that we use releases, maybe the `CHANGES.md` should just give a link to the release page for the release notes?

> In general, users take it for granted that there are breaking API changes between major versions like 0.x.y and 1.0.0, but may not between minors.

Regarding this, I'm not sure it's true ... if you look at the changelog for spire, historically there have often (always) been breaking changes between 0.x releases. Furthermore, this is common for many Scala projects using (early) semantic versioning, such as http4s, circe, etc. In fact, this is even encoded inside `sbt`, so if you try to use two different minor versions of a library in the 0.x stage it will stop you!

Reply:
> I think CHANGES.md was important before GitHub had "releases" feature. But now we that we use releases, maybe the CHANGES.md should just give a link to the release page for the release notes?

I agree. That would be better. 

Which option do you like?

1. delete duplicated migration stuffs from CHANGES.md and replace them with link to release note
2. keep migration stuffs for now and add link to release note

Reply:
in addition, If we move to github release note, I think it would be helpful to mention it at the beginning of CHANGES.md like "since v0.18.0, we track changes in \[github releases\]\(link/to/release/notes\)"

Reply:
> why aren't they the same? Is there something missing from the release notes,

AFAIK, there's nothing missing and my migration guide may be a bit redundant😓

Reply:
Yes 💯

I think the only change we need to make to `CHANGES.md` is to put a link to github releases exactly like that. That way, we avoid duplicating/separating information about changes that belongs together.

Then, I know you can't PR the release notes, but if there's important information you think is missing from there please let me know or propose some edits, I really appreciate having another pair of eyes on it :)

Thanks again for all your efforts!